src/commands/utils/purge-kernels.cc

```diff
@@ -0,0 +1,67 @@
+/*---------------------------------------------------------------------------*\
+                          ____  _ _ __ _ __  ___ _ _
+                         |_ / || | '_ \ '_ \/ -_) '_|
+                         /__|\_, | .__/ .__/\___|_|
+                             |__/|_|  |_|
+\*---------------------------------------------------------------------------*/
+
+#include "purge-kernels.h"
+#include "commands/conditions.h"
+#include "commands/commonflags.h"
+#include "utils/flags/flagtypes.h"
+#include "Zypper.h"
+#include "solve-commit.h"
+#include "global-settings.h"
+
+
+#include <zypp/Target.h>
+#include <zypp/PurgeKernels.h>
+
+using namespace zypp;
+
+PurgeKernelsCmd::PurgeKernelsCmd(std::vector<std::string> &&commandAliases_r) :
+  ZypperBaseCommand(
+    std::move( commandAliases_r ),
+    // translators: command synopsis; do not translate lowercase words
+    _("purge-kernels [OPTIONS]"),
+    // translators: command summary: targetos, tos
+    _("Remove old kernels."),
+    // translators: command description
+    _("Autoremoves installed kernels accoring to list of kernels to keep from /etc/zypp/zypp.conf:multiversion.kernels which can be given as <version>, latest(-N), running, oldest(+N)."),
+    ResetRepoManager | InitTarget | LoadTargetResolvables | Resolve
```

Comment:
Shall we introduce shortcuts? `PackageRemoveSetup = ResetRepoManager | InitTarget | LoadTargetResolvables | Resolve`, also `PackageInstallSetup`

Reply:
We could add that, but its not going to be used a lot , if you check out the code for the remove command the init is much more customized, e.g splitted in 2 parts and also uses "LoadResolvables" instead of "LoadTargetResolvables"